LayerNorm: Handling if batch size is zero

ggoossen opened the pull request:
Summary:
Handling of empty example was giving a cuda error.
Adding getLastError check to make sure cuda errors are attributed to the
correct function (instead of currently it was attributing the error to the next
cuda operator).
Added special case for batch-size zero, also added to cpu to keep things
consistent.

Resubmit of D18085429 without stacked commits

Test Plan: test included

Differential Revision: D18122212

xiaomengy reviewed (approved):
Thanks for this change.

ggoossen committed:
LayerNorm: Handling if batch size is zero (#28614)

Summary:
Handling of empty example was giving a cuda error.
Adding getLastError check to make sure cuda errors are attributed to the
correct function (instead of currently it was attributing the error to the next
cuda operator).
Added special case for batch-size zero, also added to cpu to keep things
consistent.

Resubmit of D18085429 without stacked commits
Pull Request resolved: https://github.com/pytorch/pytorch/pull/28614

Test Plan: test included

Differential Revision: D18122212

Pulled By: ggoossen

fbshipit-source-id: 43b7572b29ed3a4ade665331a4457ce01af8a828
